docstore-metrics/src/main/java/org/hypertrace/core/serviceframework/docstore/metrics/DocStoreMetricsRegistry.java

```diff
@@ -0,0 +1,152 @@
+import org.hypertrace.core.documentstore.metric.DocStoreMetric;
+import org.hypertrace.core.documentstore.metric.DocStoreMetricProvider;
+import org.hypertrace.core.documentstore.model.config.CustomMetricConfig;
+import org.hypertrace.core.serviceframework.metrics.PlatformMetricsRegistry;
+import org.hypertrace.core.serviceframework.spi.PlatformServiceLifecycle;
+
+@SuppressWarnings("unused")
+public class DocStoreMetricsRegistry {
+  private static final long INITIAL_DELAY_SECONDS = MINUTES.toSeconds(5);
+
+  private final DocStoreMetricProvider metricProvider;
+  @Nullable private PlatformServiceLifecycle platformLifecycle;
+  private int threadPoolSize;
+  private List<DocStoreCustomMetricReportingConfig> customMetricConfigs;
+  private Duration standardMetricsReportingInterval;
+
+  public DocStoreMetricsRegistry(@NonNull final Datastore datastore) {
+    metricProvider = datastore.getDocStoreMetricProvider();
+    platformLifecycle = null;
+    threadPoolSize = 1;
+    customMetricConfigs = emptyList();
+    standardMetricsReportingInterval = Duration.ofMinutes(30);
+  }
+
+  /**
+   * Supply the platform lifecycle to stop monitoring the datastore when the service is shutting
+   * down
+   */
+  public DocStoreMetricsRegistry withPlatformLifecycle(
+      final PlatformServiceLifecycle platformLifecycle) {
+    this.platformLifecycle = platformLifecycle;
+    return this;
+  }
+
+  /** Define the custom metrics to be reported */
+  public DocStoreMetricsRegistry withCustomMetrics(
+      @NonNull final List<DocStoreCustomMetricReportingConfig> customMetricConfigs) {
+    this.customMetricConfigs = customMetricConfigs;
+    return this;
+  }
+
+  /**
+   * Override the number of threads dedicated for datastore metric reporting. The default value is
+   * 1.
+   */
+  public DocStoreMetricsRegistry withThreadPoolSize(final int threadPoolSize) {
+    this.threadPoolSize = threadPoolSize;
+    return this;
+  }
+
+  /**
+   * Override the reporting interval of the standard datastore metrics. The default value is 30
+   * minutes.
+   */
+  public DocStoreMetricsRegistry withStandardMetricsReportingInterval(
+      @NonNull final Duration standardMetricsReportingInterval) {
+    this.standardMetricsReportingInterval = standardMetricsReportingInterval;
+    return this;
+  }
+
+  /**
+   * Continuously monitor a database and periodically report (standard and custom) metrics.
+   *
+   * <p>The standard metrics (like the database connection count this service is holding) are
+   * reported immediately after this method is invoked and subsequently reported at the standard
+   * metrics reporting interval (configurable using {@link #withStandardMetricsReportingInterval})
+   *
+   * <p>The custom metrics, provided through {@link #withCustomMetrics}, are reported immediately
+   * after this method is invoked and subsequently reported at the interval scheduled per metric.
+   */
+  public void monitor() {
```

Comment:
Suggest holding on to this executor. If a consumer calls monitor twice right now, it would be stuck reporting twice until server shutdown. By holding a ref to the executor you can do two things:
1. Expose a shutdown method so callers can choose to explicitly stop if not using lifecycle
2. clean up any previous executor and re-schedule if monitor is called repeatedly.